<launch>
    <arg name="pkg" default="$(find seddom)" />
    <arg name="method" default="seddom"/>
    <arg name="dataset" default="semantickitti"/>
    <arg name="debug" default="false"/>
   
    <arg name="method_config" value="$(arg pkg)/config/methods" />
    <arg name="data_config" value="$(arg pkg)/config/datasets/" />

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find seddom)/rviz/semantickitti_node.rviz" />
    <arg name="debug_prefix" value="gdb -ex run --args" if="$(arg debug)" />
    <arg name="debug_prefix" value="" unless="$(arg debug)" />
    <node pkg="seddom" type="semantickitti_node" name="semantickitti_node" output="screen" launch-prefix="$(arg debug_prefix)">

        <param name="dir" value="$(arg pkg)/data/$(arg dataset)_04" />

        <!-- Load method configuration from file -->
        <rosparam file="$(arg method_config)/$(arg method).yaml" command="load" />

        <!-- Load dataset configuration from file -->
        <rosparam file="$(arg data_config)/$(arg dataset).yaml" command="load" />
        
    </node>

</launch>
